### ISSUE TITLE
HLSL - can't initialize an empty struct

### ISSUE THREAD

**Issue by nvining-work** [KhronosGroup/glslang#862]
The following code fails to compile with glslangValidator.exe -V "VertexShader.hlsl" -e ShadeVertex -S vert -D :

---8<---

struct Test
{

};

static const Test Test_Empty;
float4 ShadeVertex(in uint vertexIndex : VERTEXID) : VS_OUT_POSITION
{ 
    return float4(0,0,0,0);
}

---8<---

If you change the Test struct:

---8<---

struct Test
{
int foo;
};

---8<---

it works.

**Comment by johnkslang** [comment]
Changing to use code blocks:

```
struct Test
{

};

static const Test Test_Empty;
float4 ShadeVertex(in uint vertexIndex : VERTEXID) : VS_OUT_POSITION
{ 
    return float4(0,0,0,0);
}
```

Works with this change:

```
struct Test
{
int foo;
};
```

**Comment by steve-lunarg** [comment]
There's a tentative fix in #867, but I'm uncertain if there's any undesired GLSL interaction.  I don't *think* so, since GLSL doesn't allow declaration of empty structs anyway, but I'm not too familiar with the GLSL FE.
